src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -177,9 +177,16 @@ private String validatePassword(final Map<String, String> propertiesSentToClient
       }
 
       return RsaAuthenticator.decryptPasswordForAction(key.get(), propertiesReadFromClient, pass -> {
+        final String legacyHashedPassword = propertiesReadFromClient.get(HASHED_PASSWORD_KEY);
```

Comment:
Any thoughts to making `propertiesReadFromClient` a strategy object?  I keep seeing examples through the code where if we made that a 1st class data object things would be much simpler. Notably much of the code here could be moved to that class, it would be a much more functional-like API and could be tested quite directly with primitives. Meanwhile it allows us to test the handling code here more easily, we can mock the `propertiesReadFromClient` and then directly simulate any of the login scenarios.

Reply:
This may seem out of scope, but I suspect this line of thinking and updates will be the key to having simple tests.

Reply:
Yeah, definetely out-of-scope, would introduce a breaking change.
Defenitely something to consider, but I'd suggest going to JSON strings instead of real objects...